src/tasks/components/Task/index.tsx

```diff
@@ -96,6 +97,9 @@ const Task = ({ data, fill = TASK_DEFAULT_FILL, onLeave, onOver, x, y, width, fi
 
   const [taskDimensions, setTaskDimensions] = useState(initialTaskDimensions);
 
+  const endDraw = useMemo(() => timeBlocks.length * columnWidth, [timeBlocks, columnWidth]);
+  const finalPoint = useMemo(() => endDraw - taskDimensions.width, [endDraw, taskDimensions]);
```

Comment:
`endDraw` is used to calculate the other variable only, you can have just `finalPoint` and do all calculations inside there.